### PR TITLE
Improve DummyLogger

### LINKED ISSUE #6395: fast_dev_run fail on log_hyperparams
## 🐛 Bug
Issue when running: `fast_dev_run=True`
"TypeError: log_hyperparams() takes 2 positional arguments but 3 were given"

### To Reproduce

When using the following: Where self.hp_metrics is a list of strings where each string is an available metric that is being logged, example "accuracy/val".
```
def on_train_start(self):
        if self.logger:
            self.logger.log_hyperparams(self.hparams, {metric:0 for metric in self.hp_metrics})
```
### Expected behavior

Assume the unit test is wrong since the documentation say that self.logger.log_hyperparams takes one positional argument and one dictionary. The code run fine without fast_dev_run=True and everything is logged correctly to tensorboard.

### Environment

pytorch_lightning 1.2.2

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #6395

This PR makes the methods in `DummyLogger` accept arbitrary positional and keyword inputs.
The new test fails on master.

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/loggers/base.py [hunk 1]
```diff
@@ -418,30 +420,29 @@ def nop(*args, **kw):
...
 
-    @rank_zero_only
-    def log_metrics(self, metrics, step):
+    def log_metrics(self, *args, **kwargs):
         pass
 
-    @rank_zero_only
-    def log_hyperparams(self, params):
+    def log_hyperparams(self, *args, **kwargs):
         pass
```
Comment (ananthsub) [inline pytorch_lightning/loggers/base.py:446]: add type hint for return value? `-> None`

File: pytorch_lightning/loggers/base.py [hunk 2]
```diff
@@ -452,7 +453,8 @@ def name(self):
     def version(self):
```
Comment (ananthsub) [inline pytorch_lightning/loggers/base.py:453]: the return values for name/version don't match the base interface

**Review by ananthsub — APPROVED** [review]

File: pytorch_lightning/loggers/base.py (hunk 2 above)
Comment (awaelchli) [reply pytorch_lightning/loggers/base.py:453]: don't know how to cleanly address this. 
one could make version and name optional in the base class, but then again they are not really optional

Comment (ananthsub) [reply pytorch_lightning/loggers/base.py:453]: For the purposes of the dummy logger, should we return empty string and 0?

**Review by Borda — APPROVED** [review]

File: pytorch_lightning/loggers/base.py [hunk 3]
```diff
@@ -279,12 +279,14 @@ def _sanitize_params(params: Dict[str, Any]) -> Dict[str, Any]:
...
     @abstractmethod
-    def log_hyperparams(self, params: argparse.Namespace):
+    def log_hyperparams(self, params: argparse.Namespace, *args, **kwargs):
         """
         Record hyperparameters.
 
         Args:
             params: :class:`~argparse.Namespace` containing the hyperparameters
+            args: Optional positional arguments, depends on the specific logger being used
+            kwargs: Optional keywoard arguments, depends on the specific logger being used
```
Comment (tchaton) [inline pytorch_lightning/loggers/base.py:289]:
```suggestion
            kwargs: Optional keyword arguments, depends on the specific logger being used
```

**Review by tchaton — APPROVED** [review]
LGTM !

**Comment by s-rog** [comment]
Tensorboardlogger on 1.2.4 is still raising the same type error.

**Comment by awaelchli** [comment]
The fix is not in 1.2.4.
Please check my repro example and compare 1.2.4 vs. master:
https://github.com/PyTorchLightning/pytorch-lightning/issues/6395#issuecomment-792342717

**Comment by awaelchli** [comment]
> I'm guessing this change needs to propogated to every logger?

I suppose yes, the interface could be standardized to match the base class signature

**Comment by s-rog** [comment]
@awaelchli this PR is listed in both the 1.2.x cherries (under 1.2.4) and in the 1.2.4 changelog as well...

I have yet to test this on master but perhaps the 1.2.4 changelog need to be updated then? and this PR added to 1.2.6 cherries?

**Comment by carmocca** [comment]
@s-rog I updated the cherries issue

**Pull request merged**